### PR TITLE
Workspace share

### PR DESCRIPTION
**Pull request by gilsonbridge**
Updated APIs for sharing workspace.

### CODE REVIEW

**Comment by BobCashStory** [comment]
@gilsonbridge to follow best practice of sonar cloud can you add `aria-hidden="true"` on all font awesome `<i>` tag ?
That will resolve 6 issue .

**Comment by BobCashStory** [comment]
I have done some fix fox bugs and code smell but it look we are for from home still ! what i propose is to add in the issue for next sprint to have a look at how we build html to never have this kind of problem again

**Review by BobCashStory — APPROVED** [review]

File: server/routes.ts [hunk 1]
```diff
@@ -113,6 +113,12 @@ export default function routes(app: Express) {
   router.route('/workspaces/:id').delete(isAuth('admin'), workspace.delete);
   router.route('/workspaces/duplicate').post(isAuth(), user.duplicateWS);
 
+  // workspace share apis
+  router.route('/workspaces/share/:id').get(isAuth(), workspace.getShare);
```
Comment (BobCashStory) [inline server/routes.ts:117]: look like every one can get the share ? not ony the creator and dest ?

File: server/routes.ts [hunk 2]
```diff
@@ -113,6 +113,12 @@ export default function routes(app: Express) {
   router.route('/workspaces/:id').delete(isAuth('admin'), workspace.delete);
```
Comment (BobCashStory) [inline server/routes.ts:113]: look like upper we have same problem every auth user can get all workspace, that not what we want

**Review by BobCashStory — CHANGES_REQUESTED** [review]

**Comment by BobCashStory** [comment]
If you found a link to a workspace who is not in your workspaces you should not get it and also see a modal that ask you the ask permission, then that send an email to owner to accept or not to share, if yes that share to the user like usually

File: server/routes.ts (hunk 1 above)
Comment (gilsonbridge) [reply server/routes.ts:96]: All the share APIs query to database with `creatorId` identified from token and hence will only work if the `creatorId` matches with the workspaceId

File: server/routes.ts (hunk 2 above)
Comment (gilsonbridge) [reply server/routes.ts:92]: Updated PR with fix in commit 6269884

**Comment by gilsonbridge** [comment]
> If you found a link to a workspace who is not in your workspaces you should not get it and also see a modal that ask you the ask permission, then that send an email to owner to accept or not to share, if yes that share to the user like usually

This feature has been implimented and updated the PR with latest commit b09c05d

**Pull request merged**